runtime/sema/type.go

```diff
@@ -3779,6 +3827,137 @@ func (t *CompositeType) GetNestedTypes() *StringTypeOrderedMap {
 	return t.NestedTypes
 }
 
+func (t *CompositeType) isTypeIndexableType() bool {
+	// resources and structs only can be indexed for attachments
+	return t.Kind == common.CompositeKindResource || t.Kind == common.CompositeKindStructure
+}
```

Comment:
ah, this could be a method on `CompositeKind` actually, then you could use it in Checker.VisitAttachExpression and Checker.VisitRemoveStatement